src/App.js

```diff
@@ -1,47 +1,172 @@
+      setCartItems(cartItems.filter((x) => x.id !== movie.id));
+    } 
+    //Otherwise decrease the movie's quantity by one
+    else {
+      setCartItems(cartItems.map((x) =>
+                   x.id === movie.id ? {...exists, quantity:exists.quantity - 1} : x));      
+    }
+    //Also remove the price from total
+    setTotalPrice(totalPrice - 10);
+  }
+
+  //---------------CART FUNCTIONALITY END--------------------
+
 
+  //---------------PAGINATION FUNCTIONALITY START----------------
+
+  //Get the movies from a specific page of the API
+  async function paginationMovies(pageNumber) {
+    try {
+      const res = await fetch(`https://api.themoviedb.org/3/search/movie?api_key=${apiKey}&language=en-US&query=${searchField}&page=${pageNumber}`);
+      const data = await res.json();
+      setMovies([...data.results]);
+    } catch (err) {
+      console.error(err);
     }
+  }
+
+  //Set the page number 
+  const nextPage = (pageNumber) => {
+    paginationMovies(pageNumber);
+    setCurrentPage(pageNumber);
+  }
+  //---------------PAGINATION FUNCTIONALITY END-----------------
+
+  //---------------EVENT HANDLERS START-----------------
+
+  //Handle the onChange event for the search input
+  const handleSearchChange = e => {
+    e.preventDefault();
+    setSearchField(e.target.value);
+    fetchSearchedMovies();
+  }
 
-    return (
-    <Layout>
+  //Handle the onChange event for the select option
+  const handleSortingChange = e => {
+    e.preventDefault();
+
+    //get the selected option
+    const sortval = e.target.value;
+    setSortType(e.target.value);
+
+    //sort the movies in a copy and then set it
+    const sorted = [...movies].sort((a, b) => b[sortval] - a[sortval]);
+    setMovies(sorted);
+  }
+  //---------------EVENT HANDLERS END-----------------
+
+
+  return (
+    <Layout countCartItems={cartItems.length}>
       <div className="App">
-       
-          <div className="row">
-            <Search/>
+
+        <div className="row">
+          <Search
+            placeholder='Search movies'
+            handleSearchChange={handleSearchChange}
+            handleSortingChange={handleSortingChange}
+            sortValue={sortType}
+          />
+        </div>
+        <div className="row">
+          <div className="col-9">
+            {movies.length === 0 ? <p className='text-white'>No movies found!</p> : <MoviesList movies={movies} onAdd={onAdd} />}
+            {totalResults > 20 ? <Pagination pages={totalPages} nextPage={nextPage} currentPage={currentPage} /> : ''}
           </div>
-          <div className="row">
-            <div className="col-9">
-              <MoviesList movies={movies} />
-            </div>
-            <div className="col-3">
-              <Cart/>
-            </div>
+          <div className="col-3">
+            <Cart 
+              cartItems={cartItems}
+              totalPrice={totalPrice}
+              onAdd={onAdd} 
+              onRemove={onRemove} 
+            />
```

Comment:
Good job, Maria! The onAdd function works correctly; my suggestion is to separate the functionalities between the add button in MovieList and the quantity increasement in the Cart component.